.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,68 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, gh-pages ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
```

Comment:
Worth analyzing on PR instead of cron? How long does this take?

Reply:
It took between 15 and 17 minutes each time I tried. That is pretty long, I removed the PR check so that it doesn't block contributors for that long. Instead now it will run once a PR is merged.

I'm open to removing the Cron it was set by default. Running it on a schedule has the benefit that it will proactively scan all the dependencies regularly so theoretically if any of them have a new security (either recently introduced to their code base or a new vulnerability added to the scanner) problem we will know earlier.

Reply:
Ah -- if it's that long, I'm totally fine with doing it periodically. On merged might be ideal, since our go mod closure shouldn't be changing outside of a merge.